lib/charm.ts

```diff
@@ -0,0 +1,90 @@
+import StatTypes from "../public/data/stat-types.json";
+
+export enum CharmPosition {
+  Left,
+  Mid,
+  Right,
+}
+
+export interface StatSummary {
```

Comment:
why an `interface` instead of a `type`? you are just defining the shape of an object and are not trying to extend this anywhere else.